### PR TITLE
[V0.1.1] - AAuser, I want SM init to recognise if i'm logged in, so that I can bypass login

### PR DESCRIPTION
**Pull request by Heyrows**
## Types of changes

- [ ] Bug fix (a non-breaking change which fixes an issue)
- [ ] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly. (PR: #)
- [ ] All new and existing tests are passing.

### CODE REVIEW

File: packages/slicemachine-core/src/core/index.ts [hunk 1]
```diff
@@ -66,4 +71,18 @@ export const Auth = {
     );
   },
   logout: (): void => removeAuthConfig(),
+  validateSession: async (
+    requiredBase: string
+  ): Promise<Communication.UserInfo | null> => {
+    console.log("validate session");
```
Comment (Miniplop) [inline packages/slicemachine-core/src/core/index.ts]: Do we want to keep this console.log ? :p

File: packages/slicemachine-core/src/core/index.ts [hunk 2]
```diff
@@ -66,4 +71,18 @@ export const Auth = {
...
+    const config: AuthConfig = getOrCreateAuthConfig();
+    console.log(config);
```
Comment (Miniplop) [inline packages/slicemachine-core/src/core/index.ts]: same question here

File: packages/slicemachine-core/src/core/index.ts [hunk 3]
```diff
@@ -66,4 +71,18 @@ export const Auth = {
...
+    if (!config.cookies.length) return Promise.resolve(null); // default config, logged out.
+    if (requiredBase != config.base) return Promise.resolve(null); // not the same base so it doesn't count.
+
+    console.log("validating now");
```
Comment (Miniplop) [inline packages/slicemachine-core/src/core/index.ts]: Re same question hihi

File: packages/slicemachine-core/__tests__/src/core/auth.test.ts [hunk 4]
```diff
@@ -69,4 +78,84 @@ describe("communication", () => {
...
+          qwerty: { dbid: "efgh", role: communication.Roles.WRITER },
+        },
+      })
+    );
+
+    const result = await Auth.validateSession(fakeBase);
+    expect(result).toBe(null);
+  });
+
+  test("validate session shouldn't work for different base", async () => {
```
Comment (Miniplop) [inline packages/slicemachine-core/__tests__/src/core/auth.test.ts]: maybe we can change the name to "validate session should return null if there is different bases"

File: packages/slicemachine-core/__tests__/src/core/auth.test.ts [hunk 5]
```diff
@@ -69,4 +78,84 @@ describe("communication", () => {
     expect(authHelpers.isHandlerData(nonHandlerData4)).toBe(false);
     expect(authHelpers.isHandlerData(nonHandlerData5)).toBe(false);
   });
+
+  test("validate session shouldn't work for empty cookies", async () => {
```
Comment (Miniplop) [inline packages/slicemachine-core/__tests__/src/core/auth.test.ts]: maybe we can change the name to "validate session should return null for empty cookies"

File: packages/slicemachine-core/__tests__/src/core/auth.test.ts [hunk 6]
```diff
@@ -69,4 +78,84 @@ describe("communication", () => {
...
+          qwerty: { dbid: "efgh", role: communication.Roles.WRITER },
+        },
+      })
+    );
+
+    const result = await Auth.validateSession(fakeBase);
+    expect(result).toBe(null);
+  });
+
+  test("validate session shouldn't work for throw in the validation", async () => {
```
Comment (Miniplop) [inline packages/slicemachine-core/__tests__/src/core/auth.test.ts]: maybe we can change the name to "validate session should return null when validate session reject the promise"

File: packages/slicemachine-core/__tests__/src/core/auth.test.ts [hunk 7]
```diff
@@ -69,4 +78,84 @@ describe("communication", () => {
...
+    });
+
+    const mockedValidate = communication.validateSession as jest.Mock;
+    mockedValidate.mockReturnValue(Promise.reject("unauthorized"));
+
+    const result = await Auth.validateSession(fakeBase);
+    expect(result).toBe(null);
+  });
+
+  test("validate session should work", async () => {
```
Comment (Miniplop) [inline packages/slicemachine-core/__tests__/src/core/auth.test.ts:140]: Is there others case where the validate session should works ?

Comment (Heyrows) [reply packages/slicemachine-core/__tests__/src/core/auth.test.ts:140]: the only other case i can think of is if the .prismic file doesn't exist but it has been covered by the getConfig which creates a new one if it doesn't exist.

**Review by Miniplop — APPROVED** [review]
Nice :)

**Pull request merged**